src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -99,15 +99,39 @@ public PipelineCacheServer(AgentTaskPluginExecutionContext context)
                     }
                     catch { }
                 }
-                
-                // Cache the artifact
-                PipelineCacheActionRecord cacheRecord = clientTelemetry.CreateRecord<PipelineCacheActionRecord>((level, uri, type) =>
-                    new PipelineCacheActionRecord(level, uri, type, PipelineArtifactConstants.SaveCache, context));
-                CreateStatus status = await pipelineCacheClient.CreatePipelineCacheArtifactAsync(options, cancellationToken, cacheRecord);
+
+                // Try to cache the artifact
+                PipelineCacheActionRecord cacheRecord = clientTelemetry.CreateRecord<PipelineCacheActionRecord>(
+                    (level, uri, type) => new PipelineCacheActionRecord(
+                        level,
+                        uri,
+                        type,
+                        PipelineArtifactConstants.SaveCache,
+                        context));
+
+                try
+                {
+                    _ = await pipelineCacheClient.CreatePipelineCacheArtifactAsync(
+                        options,
+                        cancellationToken,
+                        cacheRecord);
+                }
+                catch
+                {
+                    context.Output($"Failed to cache item.");
```

Comment:
Is there any way to capture this in our telemetry (that there was a failure and we skipped it).  We will want to track these.

Reply:
The call to `PublishTelemetry` below with the `cacheRecord` should include captured exception info.